ext/DifferentiationInterfaceEnzymeExt/reverse.jl

```diff
@@ -21,6 +30,27 @@ function DI.value_and_pullback!(
     return y, dx
 end
 
+function DI.value_and_pullback!(
+    _dx, ::AutoReverseEnzyme, f, x::X, dy::Y, extras::Nothing=nothing
+) where {X<:Number,Y<:AbstractArray}
+    y = f(x)
+    mf = Mutating(f)
+    _, new_dx = only(autodiff(Reverse, mf, Const, Duplicated(y, copy(dy)), Active(x)))
+    return y, new_dx
+end
+
+function DI.value_and_pullback!(
+    dx, ::AutoReverseEnzyme, f, x::X, dy::Y, extras::Nothing=nothing
+) where {X<:AbstractArray,Y<:AbstractArray}
+    y = f(x)
+    dx_righttype = similar(x)
+    dx_righttype .= zero(eltype(dx_righttype))
```

Comment:
Why not `dx_righttype = zero(x)` in the first place?

The variable name is also a bit weird, I would assume we always want our computations to use the right types. ;)

Reply:
it's the right type to create a `Duplicated` where `x` and `dx` are the same type

Reply:
I think the code already expresses that. 🤷‍♂️

Reply:
renamed to `dx_like_x = zero(x)`